### ISSUE TITLE
Add read/write data() access to draco::Mesh::faces_

### ISSUE THREAD

**Issue by antoine-bussy** [google/draco#641]
Hi,

Would you consider adding read/write data() access to draco::Mesh::faces_ ? That is, in the draco::Mesh class:
```c++
class Mesh
{
//.....
    PointIndex::ValueType const* face_data() const
    {
        /* Check that the data layout is coherent with pointer access */
        static_assert(std::is_integral<PointIndex::ValueType>());
        static_assert(sizeof(PointIndex::ValueType) == sizeof(PointIndex));
        static_assert(sizeof(Face) == 3 * sizeof(PointIndex));
        return reinterpret_cast<PointIndex::ValueType const*>(faces_.data());
    }

    PointIndex::ValueType* face_data()
    {
        return const_cast<PointIndex::ValueType*>(const_cast<Mesh const*>(this)->face_data());
    }
//.....
};
```
My goal is to be able to interface the Mesh class with [Eigen](http://eigen.tuxfamily.org/index.php?title=Main_Page).
For example:
```c++
#include <Eigen/Dense>

auto mesh = std::make_unique<Mesh>();
//......
using matrix_t = Eigen::Matrix3X<PointIndex::ValueType>;
mesh->SetNumFaces(1000);
auto faces = matrix_t::Map(mesh->face_data(), 3, mesh->num_faces());
faces = matrix_t::Random(3, mesh->num_faces());
//..... Use faces as a 3*N matrix
```

I can submit a PR if you're ok with it. Thanks !

**Comment by tomfinegan** [comment]
This will be present in the next release, but most likely as a protected member of draco::Mesh.

**Comment by tomfinegan** [comment]
This is included in the v1.4.3 release.
